realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -784,6 +785,82 @@ private Scanner getFullStringScanner(InputStream in) {
     }
 
     /**
+     * Bulk insert of a list of RealmObjects, this is faster than {@link #copyToRealm(Iterable)} since it
+     * doesn't return the inserted elements, and doesn't perform checks (threads, parameters etc.).
+     * <p>
+     * Please note, copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param objects RealmObjects to insert.
+     */
+    public void insertToRealm(Collection<? extends RealmModel> objects) {
+        Iterator<? extends RealmModel> iterator = objects.iterator();
+        RealmModel object = null;
+        Map<RealmModel, Long> cache = new IdentityHashMap<RealmModel, Long>(objects.size());
+        if (iterator.hasNext()) {
+            // access the first element to figure out the type
+            object = iterator.next();
+            configuration.getSchemaMediator().insertToRealm(this, object, cache);
+        }
+
+        if (iterator.hasNext()) {
+            configuration.getSchemaMediator().insertToRealm(this, object, iterator, cache);
+        }
+    }
+
+    /**
+     * Insert of a RealmObject, this is faster than {@link #copyToRealm(RealmModel)} since it
+     * doesn't return the inserted elements, and doesn't perform checks (threads, parameters etc.).
+     * <p>
+     * Please note, copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param object RealmObjects to insert.
+     */
+    public void insertToRealm(RealmModel object) {
+        Map<RealmModel, Long> cache = new IdentityHashMap<RealmModel, Long>();
+        configuration.getSchemaMediator().insertToRealm(this, object, cache);
```

Comment:
I agree to a certain point. IMO it would not be acceptable if this method could SIGSEGV for example. I think we need unit test for most common error scenarios. `null`, managed objects (this will be a copy I guess), `null` values for Required objects. Possible more?